msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -201,21 +214,95 @@ public static void create(@NonNull final Context context,
-                                                                                                       final int configFileResourceId) {
+    @WorkerThread
+    public static IPublicClientApplication create(@NonNull final Context context,
+                              final int configFileResourceId) throws InterruptedException, MsalException {
         if (context == null) {
             throw new IllegalArgumentException("context is null.");
         }
 
-        return new MultipleAccountPublicClientApplication(context, loadConfiguration(context, configFileResourceId));
+        final PublicClientApplicationConfiguration configuration = loadConfiguration(context, configFileResourceId);
+        return createPublicClientApplication(context, configuration);
     }
 
+
     /**
-     * {@link PublicClientApplication#create(Context, int, ApplicationCreatedListener)} will read the client id and other configuration settings from the
+     * {@link PublicClientApplication#createMultipleAccountPublicClientApplication(Context, int, ApplicationCreatedListener)} will read the client id and other configuration settings from the
+     * file included in your applications resources.
+     * <p>
+     * For more information on adding configuration files to your applications resources please
+     *
+     * @param context              Application's {@link Context}. The sdk requires the application context to be passed in
+     *                             {@link PublicClientApplication}. Cannot be null.
+     *                             <p>
+     *                             Note: The {@link Context} should be the application context instead of the running activity's context, which could potentially make the sdk hold a
+     *                             strong reference to the activity, thus preventing correct garbage collection and causing bugs.
+     *                             </p>
+     * @param configFileResourceId The resource ID of the raw file containing the JSON configuration for the PublicClientApplication
+     * @param listener             a callback to be invoked when the object is successfully created.
+     *                             <p>
+     *                             For more information on the schema of the MSAL config json please
+     * @param listener             a callback to be invoked when the object is successfully created.
+     * @see <a href="https://developer.android.com/guide/topics/resources/providing-resources">Android app resource overview</a>
+     * @see <a href="https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki">MSAL Github Wiki</a>
+     */
+    public static void createMultipleAccountPublicClientApplication(@NonNull final Context context,
+                              final int configFileResourceId,
+                              @NonNull final ApplicationCreatedListener listener) {
+        if (context == null) {
```

Comment:
Since `ApplicationCreatedListener` is required as well, we can throw an `IllegalArgumentException` if null

Reply:
Fixed.